src/Compilers/CSharp/Portable/Binder/IndexerArgumentInfo.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// Information about the arguments of an implicit or explicit Index or Range indexer pattern used in list patterns.
+    /// </summary>
+    internal sealed class IndexerArgumentInfo
```

Comment:
>IndexerArgumentInfo [](http://example.com/codeflow?start=26&length=19)

I'm not sure why we're using a new class here. Why aren't we just using the appropriate `BoundIndexerAccess` or `BoundCall` node?

Reply:
I previously used BindIndexer with a HasError check on BoundIndexer (see https://github.com/dotnet/roslyn/pull/53299#pullrequestreview-659398377),
but we want to bail as soon as the resolution failed. That method is just doing too many things which is specific to a source indexer access, we don't need any of that.

Reply:
I could probably follow `PerformPatternMethodLookup` logic for indexers which does exactly that.